### ISSUE TITLE
iOS 13 cannot install IPAs

### ISSUE THREAD

**Issue by asikora-sf** [microsoft/appcenter#543]
**What App Center service does this affect?**
AppCenter app installs

**Describe the bug**
iOS 13 changed the way Safari treats downloads in iPadOS (at least). They become a download to iCloud Drive instead of an install. 

This is likely a bug on Apple, and I've filed a radar there as well, but wanted to make you aware of this issue in case there are any changes that need to be done on the web-side to tell the new Safari that an app-install is desired.

**To Reproduce**
1. Install iOS 13 on an iPad
2. Go to an application view in AppCenter
3. Tap on Download
4. The app 'downloads' into the files app.

**Expected behavior**
The app should install. Alternatively Apple may allow an install action from the Files app of an ipa. 

**Screenshots**
![IMG_0005](https://user-images.githubusercontent.com/7671128/59225414-7ddc1f80-8b85-11e9-8c0d-2f0c96d1571e.jpg)

**Smartphone (please complete the following information):**
 - Device: iPad Pro first gen
 - OS: 13.0 beta 1
 - Browser: Safari

**Additional context**

**Comment by botatoes** [comment]
Thanks for filing the bug. We are aware of this and will get a fix out. This is around the introduction of iPadOS and how we are detecting devices.

**Comment by blanx** [comment]
This topic is related to the new iPadOS and the identifier of the OS

**Comment by lumaxis** [comment]
For the time being, you can use the workaround of requesting the mobile website like in the screenshot below:

![image](https://user-images.githubusercontent.com/406937/61135958-f4d94080-a4c2-11e9-94c5-152adccf02c4.png)

**Comment by Guy783** [comment]
We have a smart chap in our team and he pointed that in iOS13, in Safari settings you can set "Request Desktop Website" to false (to show the mobile site), this way when downloading an app on App Center it will show the mobile version with the "Install" button instead of "Download"

**Comment by Guy783** [comment]
> **What App Center service does this affect?**
> AppCenter app installs
> 
> **Describe the bug**
> iOS 13 changed the way Safari treats downloads in iPadOS (at least). They become a download to iCloud Drive instead of an install.
> 
> This is likely a bug on Apple, and I've filed a radar there as well, but wanted to make you aware of this issue in case there are any changes that need to be done on the web-side to tell the new Safari that an app-install is desired.
> 
> **To Reproduce**
> Steps to reproduce the behavior:
> 
> 1. Install iOS 13 on an iPad
> 2. Go to an application view in AppCenter
> 3. Tap on Download
> 4. The app 'downloads' into the files app.
> 
> **Expected behavior**
> The app should install. Alternatively Apple may allow an install action from the Files app of an ipa.
> 
> **Screenshots**
> If applicable, add screenshots to help explain your problem.
> ![IMG_0005](https://user-images.githubusercontent.com/7671128/59225414-7ddc1f80-8b85-11e9-8c0d-2f0c96d1571e.jpg)
> 
> **Smartphone (please complete the following information):**
> 
> * Device: iPad Pro first gen
> * OS: 13.0 beta 1
> * Browser: Safari
> 
> **Additional context**
> Add any other context about the problem here.



> **What App Center service does this affect?**
> AppCenter app installs
> 
> **Describe the bug**
> iOS 13 changed the way Safari treats downloads in iPadOS (at least). They become a download to iCloud Drive instead of an install.
> 
> This is likely a bug on Apple, and I've filed a radar there as well, but wanted to make you aware of this issue in case there are any changes that need to be done on the web-side to tell the new Safari that an app-install is desired.
> 
> **To Reproduce**
> Steps to reproduce the behavior:
> 
> 1. Install iOS 13 on an iPad
> 2. Go to an application view in AppCenter
> 3. Tap on Download
> 4. The app 'downloads' into the files app.
> 
> **Expected behavior**
> The app should install. Alternatively Apple may allow an install action from the Files app of an ipa.
> 
> **Screenshots**
> If applicable, add screenshots to help explain your problem.
> ![IMG_0005](https://user-images.githubusercontent.com/7671128/59225414-7ddc1f80-8b85-11e9-8c0d-2f0c96d1571e.jpg)
> 
> **Smartphone (please complete the following information):**
> 
> * Device: iPad Pro first gen
> * OS: 13.0 beta 1
> * Browser: Safari
> 
> **Additional context**
> Add any other context about the problem here.

We have a smart chap in our team and he pointed that in iOS13, in Safari settings you can set "Request Desktop Website" to false (to show the mobile site), this way when downloading an app on App Center it will show the mobile version with the "Install" button instead of "Download"

**Comment by lumaxis** [comment]
@Guy783 You're absolutely right! It's possible to change this setting for the HockeyApp website permanently:

![IMG_E79D58695F06-1](https://user-images.githubusercontent.com/406937/62822913-72f43a00-bb8a-11e9-859a-4361b6f6d34a.jpeg)

**Comment by tewner** [comment]
Hi!
Any ETA on a fix for this?

**Comment by kuyazee** [comment]
I have also observed this happening on an **iPhone 6s plus** device running on **iOS 13.1 beta**

**Comment by lumaxis** [comment]
@kuyazee This particular problem is specific to iPadOS since that changed the browser user agent that is sent to our side.
If you are experiencing issues installing on non-iPad devices, it is different from this and please open a separate Issue for that, thanks!

**Comment by anoop4real** [comment]
Any solution to this... i tried updating the settings in iPadOS safari...didnt work. The ipa is always downloaded to the iCloudDrive

**Comment by wtaconnerzeuli** [comment]
Hello! Wondering if this would affect the ability to trust enterprise developers? After installing it's telling me I need to trust my enterprise profile, but it doesn't actually have one listed under "Profiles and Device Management".

Thanks!

**Comment by marikavertzonis** [comment]
This is affecting my team also. The workaround in ok for short term but not long term.

**Comment by lumaxis** [comment]
@wtaconnerzeuli It does not, from my experience. iOS has had a long standing issue where sometimes, profiles take a while to show up in the settings page but it's definitely separate from this issue.

**Comment by Chris-Friend** [comment]
We're experiencing some variant or derivation of this issue on App Center.

It looks like a warning banner has been introduced, as referenced in [#1186](https://github.com/microsoft/appcenter/issues/1186), to warn the user to view the website in Mobile mode, but this banner is not very explicit, and too easily dismissed. It is now also not possible to download/install apps when viewing the site in Desktop mode, presumably to prevent the .ipa file from being downloaded into the device's file system.

![WarningBanner](https://user-images.githubusercontent.com/58506299/70129858-68990700-1677-11ea-9da5-12b0bad8b1a8.png)

Viewing the site on Desktop does not allow users to install apps, yet App Center only shows a dismissible warning as a solution. If the user dismisses this warning, they have restricted functionality and are none the wiser as to why.

Is it possible to more firmly require that the user access the site in Mobile mode, for example by displaying the warning until they do so, with extra information on the warning as to why and how they should heed it?

Cheers!

**Comment by cguerin** [comment]
Solution for that look for the AA in the URL bar of Safari in IOS 13, and press it..you'll see an item for "Request Desktop Website" there... that solves the issue.

**Comment by Chris-Friend** [comment]
@cguerin we are aware of the aA button and the Desktop/Mobile website options, as shown in @lumaxis's screenshots above, and how this affects the behaviour of the site. My comment was was a request for an improvement to the implemented dismissible warning to improve user experience, not a request for a solution to not being able to download/install the apps. However, thank you for taking the time to read through it.

**Comment by triplef** [comment]
Is there any update on this? Our team is running into this on a regular basis, esp. when onboarding new members. Would be very nice if we didn’t have to waste that time. :)
